src/main/java/sign_up/drivers/SignUpDataAccess.java

```diff
@@ -0,0 +1,207 @@
+package sign_up.drivers;
+
+import com.opencsv.CSVWriter;
+import com.opencsv.exceptions.CsvValidationException;
+import entities.User;
+import sign_up.use_case.DsGateway;
+import com.opencsv.CSVReader;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class SignUpDataAccess implements DsGateway {
+    private final String userFilePath;
+    private final String adminFilePath;
+    private final String numUsersCreatedFilePath;
+
+    public SignUpDataAccess(String userFilePath, String adminFilePath, String numUsersCreated) {
+        this.userFilePath = userFilePath;
+        this.adminFilePath = adminFilePath;
+        this.numUsersCreatedFilePath = numUsersCreated;
+    }
+
+    @Override
+    public int getNumberUsers() {
+        ArrayList<String[]> numUsers;
+        try {
+            numUsers = readLineByLine(numUsersCreatedFilePath);
+        } catch(IOException e) {
+            System.out.println("Wrong Path");
+            String[] defaultEntry = {"0"};
+            numUsers = (ArrayList<String[]>) Arrays.asList(new String[][]{defaultEntry});
+        } catch(CsvValidationException e) {
```

Comment:
Maybe just catch `CsvException` instead. I'm not too familiar with the `CsvException` hierarchy and what subclasses extend it (and can be thrown while parsing a csv file) but it might just be safer to catch the parent exception.

Reply:
Also applies elsewhere where `CsvValidationException` is caught

Reply:
Changed this